File: src/main/java/net/ndrei/teslacorelib/items/EnergyUpgradeTier2.java
package net.ndrei.teslacorelib.items;

import com.google.common.collect.Lists;
import net.minecraft.item.ItemStack;
import net.minecraft.item.crafting.IRecipe;
import net.minecraftforge.oredict.ShapedOreRecipe;
import net.ndrei.teslacorelib.TeslaCoreLib;

import java.util.List;

/**
 * Created by CF on 2017-04-08.
 */
public class EnergyUpgradeTier2 extends EnergyUpgrade {
    public EnergyUpgradeTier2() {
        super(2);
    }

    @Override
    protected List<IRecipe> getRecipes() {
        List<IRecipe> recipes = Lists.newArrayList();

        recipes.add(new ShapedOreRecipe(new ItemStack(this, 1),
                " g ", "rbr", "rrr",
                'b', TeslaCoreLib.baseAddon,
                'r', "dustRedstone", // Items.REDSTONE,
                'g', "gearDiamond" // TeslaCoreLib.gearDiamond)
        ));

        recipes.add(new ShapedOreRecipe(new ItemStack(this, 1),
                " d ", "dbd", " d ",
                'b', TeslaCoreLib.energyUpgradeTier1,
                'd', "gemDiamond" // Items.DIAMOND
        ));

        return recipes;
    }
}

File: src/main/kotlin/net/ndrei/teslacorelib/items/EnergyUpgradeTier2.kt
package net.ndrei.teslacorelib.items

import net.ndrei.teslacorelib.ProxyLoadLevel
import net.ndrei.teslacorelib.TeslaCoreLib
import net.ndrei.teslacorelib.annotations.AutoRegisterItem
import net.ndrei.teslacorelib.config.TeslaCoreLibConfig

/**
 * Created by CF on 2017-06-27.
 */
@AutoRegisterItem(TeslaCoreLibConfig.REGISTER_ADDONS, TeslaCoreLibConfig.REGISTER_ENERGY_ADDONS)
object EnergyUpgradeTier2 : EnergyUpgrade(2) {
    init {
        TeslaCoreLib.proxy.testLoadLevel(ProxyLoadLevel.ITEMS, this.javaClass.simpleName)
    }
}
